allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_ecephys_project_cloud_api.py

```diff
@@ -0,0 +1,336 @@
+
+        Returns
+        -------
+        BehaviorProjectCloudApi instance
+
+        """
+        cache = S3CloudCache(cache_dir,
+                             bucket_name,
+                             project_name,
+                             ui_class_name=ui_class_name)
+        return cls(cache)
+
+    @staticmethod
+    def from_local_cache(
+        cache_dir: Union[str, Path],
+        project_name: str,
+        ui_class_name: str,
+        use_static_cache: bool = False
+    ) -> "BehaviorEcephysProjectCloudApi":
+        """instantiates this object with a local cache.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        ui_class_name: str
+            Name of user interface class (used to populate error messages)
+
+        Returns
+        -------
+        BehaviorProjectCloudApi instance
+
+        """
+        if use_static_cache:
+            cache = StaticLocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        else:
+            cache = LocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        return BehaviorEcephysProjectCloudApi(cache, local=True)
+
+    def get_behavior_session(
+            self, behavior_session_id: int) -> BehaviorSession:
+        """get a BehaviorSession by specifying behavior_session_id
+
+        Parameters
+        ----------
+        behavior_session_id: int
+            the id of the behavior_session
+
+        Returns
+        -------
+        BehaviorSession
+
+        Notes
+        -----
+        behavior session does not include file_id.
+        The file id is accessed via ecephys_session_id key
+        from the ecephys_session_table
+        """
+        row = self._behavior_session_table.query(
+                f"behavior_session_id=={behavior_session_id}")
+        if row.shape[0] != 1:
+            raise RuntimeError("The behavior_session_table should have "
+                               "1 and only 1 entry for a given "
+                               "behavior_session_id. For "
+                               f"{behavior_session_id} "
+                               f" there are {row.shape[0]} entries.")
+        row = row.squeeze()
+        esid = row.ecephys_session_id
+        row = self._ecephys_session_table.query(f"index=={esid}")
+        file_id = str(int(row[self.cache.file_id_column]))
+        data_path = self._get_data_path(file_id=file_id)
+        return BehaviorSession.from_nwb_path(str(data_path))
```

Comment:
If you are instantiating from an ecephys_session_id, why are you not returning a `BehaviorEcephysSession`? I don't see any reason to not return the full session data if the user ends up requesting a behavior session that is actually a behavior + ecephys session.

Reply:
I thought that if users wants the full data, their would call `get_ecephys_session` instead. Otherwise, why have `get_behavior_session` at all?

Reply:
Technically, since we aren't releasing behavior-only data, I am in favor of not having `get_behavior_session` yet. We can add that when we release the behavior-only data.